viewer/packages/camera-manager/src/Keyboard.ts

```diff
@@ -46,51 +55,67 @@ export default class Keyboard {
     this.addEventListeners();
   }
 
-  public isPressed(key: string): boolean {
-    return this.keys[key] >= 1;
+  public isPressed(key: EventCode): boolean {
+    return this._keys.has(key);
+  }
+
+  public isShiftPressed(): boolean {
+    return this.isPressed('ShiftLeft') || this.isPressed('ShiftRight');
+  }
+
+  public isCtrlPressed(): boolean {
+    return this.isPressed('ControlLeft') || this.isPressed('ControlRight');
+  }
+
+  public isAltPressed(): boolean {
+    return this.isPressed('AltLeft') || this.isPressed('AltRight');
+  }
+
+  public isMetaPressed(): boolean {
+    return this.isPressed('MetaLeft') || this.isPressed('MetaRight');
   }
 
   public dispose(): void {
     this.clearPressedKeys();
     this.removeEventListeners();
   }
 
+  /**
+   * Calculates the movement direction based on keyboard input.
+   *
+   * Given two keys representing positive and negative directions (e.g., left and right, or up and down),
+   * this method returns 1 for the positive key, -1 for the negative key, or 0 if neither or both is pressed.
+   *
+   * @param positiveKey - Key representing the positive movement direction.
+   * @param negativeKey - Key representing the negative movement direction.
+   * @returns A value indicating the direction of movement: 1 (positive), -1 (negative), or 0 (none).
+   */
+  public getKeyboardMovementValue(positiveKey: EventCode, negativeKey: EventCode): number {
+    return (this.isPressed(positiveKey) ? 1 : 0) - (this.isPressed(negativeKey) ? 1 : 0);
+  }
+
   private readonly addEventListeners = () => {
     this.clearPressedKeys();
-
-    this._domElement.addEventListener('keydown', this.onKeydown);
-    this._domElement.addEventListener('keyup', this.onKeyup);
+    this._domElement.addEventListener('keydown', this.onKeyDown);
+    this._domElement.addEventListener('keyup', this.onKeyUp);
     this._domElement.addEventListener('blur', this.clearPressedKeys);
   };
 
   private readonly removeEventListeners = () => {
-    this._domElement.removeEventListener('keydown', this.onKeydown);
-    this._domElement.removeEventListener('keyup', this.onKeyup);
+    this._domElement.removeEventListener('keydown', this.onKeyDown);
+    this._domElement.removeEventListener('keyup', this.onKeyUp);
     this._domElement.removeEventListener('blur', this.clearPressedKeys);
   };
 
-  private readonly onKeydown = (event: KeyboardEvent) => {
-    if (event.metaKey || event.altKey || event.ctrlKey) {
-      return;
-    }
-
-    if (event.keyCode in keyMap) {
-      if (this.keys[keyMap[event.keyCode]] === 0) {
-        this.keys[keyMap[event.keyCode]] = 2;
-      }
-      event.preventDefault();
```

Comment:
I think we should include this still, not sure if it's entirely necessary, but I'd prefer to break as little of the behavior as necessary. That probably means we have to check for `metaKey` etc. again too, since preventing default-behavior for those buttons disable most browser key shortcuts, i believe.

Reply:
Are you sure you want to keep it? I am pretty certain that removing this won't have any practical differences in behavior, and adding it back makes the code quite a lot less nice IMO. This is what ChatGPT says about it:

![image](https://github.com/cognitedata/reveal/assets/7979907/52757f27-10ac-489f-ac4b-e9069d0ad41c)

(😅 I added the word "legacy" to nudge ChatGPT to say: "That can probably be deleted")

Reply:
Okay, I'll accept it - We'll hear soon enough if this messes with someone :P